app/js/lib/foxbox/services/base.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const TYPE = 'Unknown';
+
+const p = Object.freeze({
+  // Private properties.
+  net: Symbol('net'),
+  settings: Symbol('settings'),
+
+  // Public getter only properties.
+  id: Symbol('id'),
+  getters: Symbol('getters'),
+  setters: Symbol('setters'),
+
+  // Private methods.
+  getChannel: Symbol('getChannel'),
+  getSetterValueType: Symbol('getSetterValueType'),
+});
+
+export default class BaseService {
+  constructor(props, config) {
+    // Private properties.
+    this[p.net] = config.net;
+    this[p.settings] = config.settings;
+
+    // Public getter only properties.
+    this[p.id] = props.id;
+    this[p.getters] = props.getters;
+    this[p.setters] = props.setters;
+
+    // Public properties.
+    this._tags = props.tags;
+  }
+
+  // Public getter only properties.
+  get type() {
+    return TYPE;
+  }
+
+  get id() {
+    return this[p.id];
+  }
+
+  get name() {
+    return '';
+  }
+
+  get hasGetters() {
+    return Object.keys(this[p.getters]).length > 0;
+  }
+
+  get hasSetters() {
+    return Object.keys(this[p.setters]).length > 0;
+  }
+
+  // Public methods.
+  /**
+   * Call a service setter with a value.
+   *
+   * @param {Object} selector
```

Comment:
I mean, the type of parameter is Object|string and description of the parameters would make it clearer what this ```selector``` is for.